scripts/backend.py

```diff
@@ -169,6 +238,312 @@ def page_not_found(error):
     return send_file(in_http_dir("404.html"))
 
 
+# streaming stuff begins
+# streaming overrides cyberia? (configure this in DEFAULT_LIVESTREAM_INFO)
+
+@app.route("/schedule.json", methods=["GET"])
+def schedule():
+    dt = datetime.today()
+    start = dt - timedelta(days=dt.weekday())
+    end = start + timedelta(days=6)
+    week_of = "{} - {}".format(start.strftime('%b %d'), end.strftime('%b %d'))
+    sched_info = {'week_of': week_of}
+    sched_path = os.path.join(db.SAVEDATA_PATH, 'schedule.txt')
+    if os.path.exists(sched_path):
+        with open(sched_path, 'r', encoding='utf-8') as sched_file:
+            sched_lines = sched_file.read()
+            sched_lines = sched_lines.split('\n')
+            sched_lines = [line for line in sched_lines if not line.startswith('//')]
+            sched_info['dailies'] = sched_lines
+    resp = make_response(json.dumps(sched_info), 200)
+    resp.headers["Content-Type"] = "application/json"
+    return resp
+
+
+def start_streaming():
+    # right now, this immediately stop the stream if you call actually_start_streaming instead
+    # this can wait for current track to finish playing if you uncomment below instead 
+    actually_start_streaming()
+    return
+
+    # try:
+    #     client = MPDClient()
+    #     client.connect(args["--mpd-host"], LIVESTREAM_INFO['PORT'])
+    # except:
+    #     print('couldn\'t connect to mpd')
+    #     return
+
+    # pinfo = get_playlist_info(client, 0, 0, True)
+    # wait_for_how_long = float(pinfo['current']['time']) - float(pinfo['elapsed'])
+    # threading.Timer(wait_for_how_long, actually_start_streaming).start()
+
+
+def actually_start_streaming():
+    # need to remake the client otherwise
+    # mpd.ConnectionError: Connection lost while reading line
+    try:
+        client = MPDClient()
+        client.connect(args["--mpd-host"], LIVESTREAM_INFO['PORT'])
+    except:
+        print('couldn\'t connect to mpd')
+        return
+
+    client.pause(1)
```

Comment:
I've had some more thoughts about this, and I guess MPD doesn't actually need to be paused and its outputs disabled.  It can just keep playing while the stream is happening (people will be able to access it at `/channel-cyberia.mp3` or something if they want).

Icecast can handle switching between the two audio sources automatically, which then removes the need for this logic.